java/src/main/java/com/google/appengine/tools/pipeline/impl/model/JobInstanceRecord.java

```diff
@@ -44,11 +45,11 @@
   // transient
   private Job<?> jobInstance;
 
-  public JobInstanceRecord(JobRecord job, Job<?> jobInstance) {
+  public JobInstanceRecord(JobRecord job, Job<?> jobInstance, JobSetting.DisplayName displayNameSetting) {
     super(job.getRootJobKey(), job.getGeneratorJobKey(), job.getGraphGuid());
     jobKey = job.getKey();
     jobClassName = jobInstance.getClass().getName();
-    jobDisplayName = jobInstance.getJobDisplayName();
+    jobDisplayName = (displayNameSetting == null ? jobInstance.getJobDisplayName() : displayNameSetting.getValue());
```

Comment:
This is my biggest comment. I think that if a Job overrides getDisplayName, the overridden method return value should be used. I think the way to achieve that is to:
1) Keep displayName as a persistent value in JobRecord.
2) When Job is given to JobInstanceRecord make sure it is set with JobRecord (and do it later on in the constructor).
3) default implementation of Job#getDisplayName() would be to check if its associated jobRecord (it has it as a variable) #getDisplayName (add this method) returning a value which is not null otherwise returns getClass().getName()

What do you think?